### PR TITLE
PDASHOSS01-47 enable UI of moving issue to another project

### PR DESCRIPTION
**Pull request by irichard00**
## What & why

Adds a **"Move to project"** action to the web UI so a user can move a work item into another project in the same workspace — closing PDASHOSS01-47.

The capability already existed via the **API-key v1 endpoint** (and the `pidash issue move <project>` CLI, already on `main`). But the web app authenticates with a **session cookie** against `pi_dash.app.urls`, and the v1 endpoint only accepts `APIKeyAuthentication`, so the browser could not reach it. This PR adds the missing session-authed endpoint and the UI on top of it.

Cross-workspace moves are intentionally out of scope — the picker only lists projects in the current workspace.

## Backend

- Extract the move logic out of `IssueMoveAPIEndpoint` into a shared `pi_dash.utils.issue_move.move_work_item_to_project(...)` helper (plus an `IssueMoveError` for the 400/403 cases). The v1 endpoint now delegates to it, so its behavior is unchanged.
- Add a session-authed `IssueMoveEndpoint` (`pi_dash/app/views/issue/move.py`) routed at:
  `POST /api/workspaces/<slug>/projects/<project_id>/work-items/<pk>/move/`
- `@allow_permission([ADMIN, MEMBER])` guards the source project; the shared helper additionally verifies the acting user is a member of the **target** project.

## Frontend

- `IssueService.moveIssue()` → posts to the app move endpoint.
- Base issues store `moveIssue()` — on success drops the work item from the source project's caches (it now lives elsewhere), mirroring the delete flow.
- `MoveIssueModal` — a searchable project picker listing same-workspace projects the user belongs to (excluding the current one). On success it toasts and navigates to the work item in its new project.
- A **"Move to project"** item added to the work item detail quick-actions (ellipsis / right-click) menu, shown to users with edit permission.

## Testing

- New contract tests `pi_dash/tests/contract/app/test_move_endpoint.py` (200 success + project reassignment, 400 missing target, 403 non-member target). The **existing** v1 contract test guards the refactor.
- Ran locally against a real Postgres: **4 move tests pass** (3 new app + 1 existing v1).
- `manage.py check` → 0 issues (URLconf + imports load, no import cycle).
- `ruff check` clean on all changed Python (removed 17 imports orphaned by the extraction).
- Web `check:types` passes; `oxlint` on new file is clean; `oxfmt` applied.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/web/core/components/issues/move-issue-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  const targetProjectIds = useMemo(
+    () => joinedProjectIds.filter((id) => id !== projectId),
+    [joinedProjectIds, projectId]
+  );
+
+  const filteredProjectIds = targetProjectIds.filter((id) => {
+    const project = getProjectById(id);
+    const projectQuery = `${project?.identifier ?? ""} ${project?.name ?? ""}`.toLowerCase();
+    return projectQuery.includes(query.toLowerCase());
+  });
```
[the remark this answers is not in the document]
Comment (irichard00) [reply apps/web/core/components/issues/move-issue-modal.tsx:74]: Confirmed and fixed in 3176a095 — unresolved projects are now dropped from `filteredProjectIds`, so an unhydrated project can no longer match every query and suppress the empty state.

File: apps/api/pi_dash/app/views/issue/move.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    def post(self, request, slug, project_id, pk):
+        try:
+            issue = move_work_item_to_project(
+                slug=slug,
+                project_id=project_id,
+                pk=pk,
+                target_ref=request.data.get("project"),
+                actor=request.user,
+                origin=base_host(request=request, is_app=True),
+            )
```
[the remark this answers is not in the document]
Comment (irichard00) [reply apps/api/pi_dash/app/views/issue/move.py:42]: Confirmed and fixed in 3176a095 — a non-dict body is now coerced to `{}`, so a malformed payload returns a clean 400 ("project is required") via the shared helper instead of an AttributeError 500.

File: apps/api/pi_dash/api/views/issue.py [hunk 3]
```diff
@@ -872,153 +858,18 @@ class IssueMoveAPIEndpoint(BaseAPIView):
...
-                **update_kwargs
+        try:
+            issue = move_work_item_to_project(
+                slug=slug,
+                project_id=project_id,
+                pk=pk,
+                target_ref=request.data.get("project"),
+                actor=request.user,
+                origin=base_host(request=request, is_app=True),
             )
```
[the remark this answers is not in the document]
Comment (irichard00) [reply apps/api/pi_dash/api/views/issue.py:873]: Confirmed and fixed in 3176a095 — same non-dict guard applied here on the v1 (API-key) endpoint.

**Comment by irichard00** [comment]
## Review summary — automated review pass

Reviewed the full diff (backend move-logic extraction + session-authed endpoint, frontend service/store/modal/menu wiring, contract tests).

**Overall: solid.** The extraction into `move_work_item_to_project` keeps the v1 (API-key) and app (session) surfaces in lockstep, `origin=base_host(is_app=True)` matches the pre-refactor v1 behavior, the store's optimistic drop mirrors the existing `removeIssue` pattern, and post-move navigation is correct (the work item keeps its pk). Contract tests cover 200 / 400 / 403 on the app endpoint.

**Findings — all confirmed, all now fixed in `3176a095`:**

1. **Modal blank-area edge case** (`move-issue-modal.tsx`) — an unhydrated project (`getProjectById` → `undefined`) produced a query string of `" "` that matched every filter, then rendered as `null`; if *all* candidates were unresolved the empty-state message was suppressed, leaving a blank panel. Fixed by dropping unresolved projects from `filteredProjectIds`.
2. **Non-dict payload → 500** (both `app/views/issue/move.py` and `api/views/issue.py`) — a malformed JSON body (list/primitive) has no `.get`, raising `AttributeError` (500). Both endpoints now coerce a non-dict body to `{}`, so it flows into the existing `"project is required"` **400** path.

No larger correctness, security, or behavior-drift issues found. Validation on the fixes: `ruff check` clean, `oxlint` clean on the modal, Python syntax OK. The valid-body path is byte-identical, so existing contract tests continue to apply.

*Posted by the automated review pass for PDASHOSS01-47.*
